### ISSUE TITLE
Deploy website via gh-pages branch rather than /docs

### ISSUE THREAD

**Issue by viglucci** [rsocket/rsocket-website#36]
## Motivation

GItHub pages allows for deployment of static files served either from a `/docs` directory or from a special `gh-pages` branch. I believe that deploying via a `gh-pages` directory is more efficient than a `/docs` directory as it allows us to not need to check in the built static output to version control, as well as avoids issues such as #30.

## Desired solution

Repository and deployment configured to leverage `gh-pages` branch rather than `/docs` output dir.

## Considered alternatives

Continue to deploy via `/docs`.

## Additional context

Deploying from `gh-pages` would have helped avoid issues like #30.

**Comment by flyinprogrammer** [comment]
> I believe that deploying via a gh-pages directory is more efficient than a /docs directory as it allows us to not need to check in the built static output to version control

I don't believe is true. It's my understanding that the static content would still get published to the branch, it's just that the static content would be at the root of the branch: https://docs.github.com/en/github/working-with-github-pages/creating-a-github-pages-site

But it wooooullld remove the confusion of `docs` vs `content-docs` :upside_down_face:

**Comment by viglucci** [comment]
Indeed. I guess my point being that there would be a dedicate branch that houses the compiled output of the site, which would help with the confusion as your noted. It also helps with PRs as we don't need to mentally ingore the compiled output and could then just look at the modified source for the PR.

**Comment by viglucci** [comment]
I don't see any immediate issues with using a new repo for the deployments as you've suggested. When discussing with @flyinprogrammer he detailed a few benefit to that strategy as well:

- Seperate ref logs for vcs related to source commits and deployment commits
- Enables the `rsocket.github.io` domain should we need it someday

**Comment by flyinprogrammer** [comment]
> Is my understanding correct that having a repository named rsocket.github.io would automatically expose the site at that domain (using the contents of the rsocket.github.io repository)?

I think you're right, I think hosting it in rsocket.github.io would trash our SEO.
Funny enough the DNS CNAME for www.rsocket.io points to rsocket.github.io, which then gets redirected to our site rsocket.io

```
dig +short www.rsocket.io
rsocket.github.io.
185.199.110.153
```

```
curl --HEAD www.rsocket.io
HTTP/1.1 301 Moved Permanently
Server: GitHub.com
Content-Type: text/html
Location: https://rsocket.io/
```

which is working as intended: https://docs.github.com/en/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain

My vote is we just use https://github.com/rsocket/rsocket-website/pull/47 to move to gh-pages and stop painting bike sheds :kissing:

**Comment by viglucci** [comment]
> My vote is we just use #47 to move to gh-pages and stop painting bike sheds 😗

+1

**Comment by viglucci** [comment]
Closed by #47
